MockDuck/Sources/MockRequest.swift

```diff
@@ -13,6 +13,26 @@ import Foundation
 final class MockRequest {
     var request: URLRequest
 
+    private(set) lazy var normalizedRequest: URLRequest = {
+        return MockDuck.delegate?.normalizedRequest(for: request) ?? request
+    }()
+
+    var serializedHashValue: String {
+        let normalizedRequest = self.normalizedRequest
+
+        var hashData = Data()
+
+        if let urlData = normalizedRequest.url?.absoluteString.data(using: .utf8) {
+            hashData.append(urlData)
+        }
+
+        if let body = normalizedRequest.httpBody {
+            hashData.append(body)
```

Comment:
Also, this body is typically created using a dictionary, which won't guarantee the order across multiple runs which will cause a different hash to be calculated. So we also added a sorting function and conversion to a string to generate the hash. This way, both the recording and the loading  reproduce the same hash every time.

```swift
func sort(dictionary: [String: Any]) -> String {
    let sorted = dictionary.keys.sorted().reduce("") { (acc, iteration) -> String in
        if let json = dictionary[iteration] as? [String: Any] {
            return acc.appending("\(iteration):\(sort(dictionary: json)),")
        } else {
            return acc.appending("\(iteration):\(dictionary[iteration]!),")
        }
    }
    return String(sorted.dropLast())
}
```

Reply:
See #24